src/Components/ArticlePage.js

```diff
@@ -0,0 +1,56 @@
+import "./ArticlePage.css";
+import { useParams, useLocation } from "react-router-dom";
+import { getArticleById } from "../Api/Api";
+import React, { useEffect, useState } from "react";
+
+const ArticlePage = (props) => {
+  const { id } = useParams();
+  const { state } = useLocation();
+  const [article, setArticle] = useState(
+    state?.article || props.article || null
```

Comment:
This could then be cleaned up - you should never be setting a value received on props in state - you already have access